operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeploymentDependentResource.java

```diff
@@ -439,14 +445,18 @@ private void addEnvVars(StatefulSet baseDeployment, Keycloak keycloakCR, TreeSet
         LinkedHashMap<String, EnvVar> varMap = Stream.concat(Stream.concat(env.stream(), firstClasssEnvVars.stream()), additionalEnvVars.stream())
                 .collect(Collectors.toMap(EnvVar::getName, Function.identity(), (e1, e2) -> e1, LinkedHashMap::new));
 
-        String truststores = SERVICE_ACCOUNT_DIR + "ca.crt";
+        String truststores = SERVICE_ACCOUNT_DIR;
+        if (baseDeployment.getSpec().getTemplate().getSpec().getAutomountServiceAccountToken()) {
+            truststores += "ca.crt";
+        }
 
         if (useServiceCaCrt) {
             truststores += "," + SERVICE_CA_CRT;
         }
-
-        // include the kube CA if the user is not controlling KC_TRUSTSTORE_PATHS via the unsupported or the additional
-        varMap.putIfAbsent(KC_TRUSTSTORE_PATHS, new EnvVarBuilder().withName(KC_TRUSTSTORE_PATHS).withValue(truststores).build());
+        if (Boolean.TRUE.equals(baseDeployment.getSpec().getTemplate().getSpec().getAutomountServiceAccountToken()) || useServiceCaCrt) {
+            // include the kube CA if the user is not controlling KC_TRUSTSTORE_PATHS via the unsupported or the additional
+            varMap.putIfAbsent(KC_TRUSTSTORE_PATHS, new EnvVarBuilder().withName(KC_TRUSTSTORE_PATHS).withValue(truststores).build());
+        }
```

Comment:
```suggestion
        if (!Boolean.FALSE.equals(keycloakCR.getSpec().getAutomountServiceAccountToken())) {
            String truststores = SERVICE_ACCOUNT_DIR + "ca.crt";

            if (useServiceCaCrt) {
                truststores += "," + SERVICE_CA_CRT;
            }
        
            // include the kube CA if the user is not controlling KC_TRUSTSTORE_PATHS via the unsupported or the additional
            varMap.putIfAbsent(KC_TRUSTSTORE_PATHS, new EnvVarBuilder().withName(KC_TRUSTSTORE_PATHS).withValue(truststores).build());
        }
```

Switching to using the entire service account directory if automount is disabled doesn't seem like the right thing to do. How about updating the truststores only if automount is enabled.